packages/pwa-kit-extension-sdk/src/react/components/withApplicationExtensionStore.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Third-Party
+import React from 'react'
+
+// Local
+import {useApplicationExtensionsStore} from '../hooks/useApplicationExtensionsStore'
+
+// Local Types
+import {SliceInitializer} from '../hooks/useApplicationExtensionsStore'
+
+// Local Types
+type withStoreOptions = {
+    id: string
+    initializer: SliceInitializer<any>
+}
+
+/**
+ * This HOC is used to add a slice to the global store for an extension.
+ *
+ * @param WrappedComponent
+ * @param options
```

Comment:
it would be nice to have types for id and initializer

Reply:
Are you talking about in the JSDoc? because there are already typescript types.... 

On a side note, I think I'm going to change the options argument to be called "config" so it's not implied that it can be optional.

Reply:
I've gone ahead and updated the file. Please take a look when you have time.